Dockerfile

```diff
@@ -1,8 +1,71 @@
-FROM node:18
-RUN git clone https://github.com/jakecoppinger/matrix-chatgpt-bot.git
-WORKDIR matrix-chatgpt-bot
-COPY env .env
-#RUN yarn install
-RUN yarn install --har --production=true
+FROM node:18 as builder
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Install app dependencies
+COPY package*.json ./
+
+# We don't need the standalone Chromium here.
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+
+RUN yarn install --frozen-lockfile && yarn cache clean
+
+COPY . .
+
 RUN yarn build
-RUN yarn start
+
+
+FROM node:18-slim
+
+RUN mkdir -p /home/pptruser/Downloads
+# Create app directory
+WORKDIR /home/pptruser
+
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium
+ENV CHROME_PATH=/usr/bin/chromium
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt update -qq \
+    && apt install -qq -y --no-install-recommends \
+      curl \
+      git \
+      gnupg \
+      libgconf-2-4 \
+      libxss1 \
+      libxtst6 \
+      python \
+      g++ \
+      build-essential \
+      chromium \
+      chromium-sandbox \
+      dumb-init \
+      fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /src/*.deb
+
+# To run Headful mode, you will need to have a display, which is not present in a server.
+# To avoid this, we will use Xvfb, and create a fake display, so the chrome will think there is a display and run properly.
+# So we just need to install Xvfb and Puppeteer related dependencies.
```

Comment:
Ah I didn't see this message yet. I haven't used Xvfb before - does it make a virtual "display" which you can then forward over SSH?

Reply:
It is a fake X server, don't think you can forward over SSH but you shouldn't need to.